src/Controller/ControllerFactory.php

```diff
@@ -249,6 +249,18 @@ protected function coerceStringToType(string $argument, ReflectionNamedType $typ
         switch ($type->getName()) {
             case 'string':
                 return $argument;
+            case 'float':
+                if (is_numeric($argument)) {
+                    return (float)$argument;
+                }
+                break;
+            case 'int':
+                if (is_numeric($argument)) {
+                    return (int)$argument;
+                }
+                break;
+            case 'bool':
+                return (bool)$argument;
```

Comment:
I think we should have an allowed list of boolean values here. I propose we only cast `'1', '0'` into booleans. This will match the behavior we have in `Validation::boolean()`.

Reply:
There is also: https://www.php.net/manual/en/filter.filters.validate.php

> Returns true for "1", "true", "on" and "yes". Returns false otherwise.

Reply:
That could work well. We don't accept on/off, yes/no when we validate boolean elsewhere though. Having consistent behavior with Validation would be nice.

Reply:
Updated to allow only `0` and `1` and included test for failure.